lib/lightning/kafka_triggers/message_candidate_set_supervisor.ex

```diff
@@ -44,4 +49,52 @@ defmodule Lightning.KafkaTriggers.MessageCandidateSetSupervisor do
       |> Supervisor.child_spec(id: "mcs_worker_#{index}")
     end)
   end
+
+  def generate_child_specs(:message_candidate_set, number_of_workers) do
+    no_set_delay =
+      Application.get_env(:lightning, :kafka_triggers)[
+        :no_message_candidate_set_delay_milliseconds
+      ]
+
+    next_set_delay =
+      Application.get_env(:lightning, :kafka_triggers)[
+        :next_message_candidate_set_delay_milliseconds
+      ]
+
+    workers =
+      0..(number_of_workers - 1)
+      |> Enum.map(fn index ->
+        {
+          MessageCandidateSetWorker,
+          [no_set_delay: no_set_delay, next_set_delay: next_set_delay]
+        }
+        |> Supervisor.child_spec(id: "mcs_worker_#{index}")
+      end)
+
+    [MessageCandidateSetServer] ++ workers
+  end
+
+  def generate_child_specs(:message, number_of_workers) do
+    no_set_delay =
+      Application.get_env(:lightning, :kafka_triggers)[
+        :no_message_candidate_set_delay_milliseconds
+      ]
+
+    next_set_delay =
+      Application.get_env(:lightning, :kafka_triggers)[
+        :next_message_candidate_set_delay_milliseconds
+      ]
+
+    workers =
+      0..(number_of_workers - 1)
+      |> Enum.map(fn index ->
+        {
+          MessageWorker,
+          [no_set_delay: no_set_delay, next_set_delay: next_set_delay]
+        }
+        |> Supervisor.child_spec(id: "message_worker_#{index}")
+      end)
+
+    [MessageServer] ++ workers
```

Comment:
It is okay but `[MessageServer | workers]` is more idiomatic for head and tail

Reply:
@jyeshe I have corrected this in a new PR.